go.mod

```diff
@@ -6,34 +6,41 @@ require (
 	github.com/BurntSushi/toml v0.3.1
 	github.com/Masterminds/semver/v3 v3.1.1
 	github.com/aryann/difflib v0.0.0-20170710044230-e206f873d14a
-	github.com/aws/aws-sdk-go v1.31.12 // indirect
 	github.com/buildpacks/imgutil v0.0.0-20210315155240-52098da06639
 	github.com/buildpacks/lifecycle v0.10.2
 	github.com/ghodss/yaml v1.0.0
-	github.com/go-openapi/spec v0.19.9
+	github.com/go-openapi/spec v0.20.3
 	github.com/google/go-cmp v0.5.6
 	github.com/google/go-containerregistry v0.6.0
 	github.com/google/go-containerregistry/pkg/authn/k8schain v0.0.0-20210322164748-a11b12f378b5
 	github.com/jinzhu/gorm v1.9.12 // indirect
 	github.com/libgit2/git2go/v31 v31.4.14
 	github.com/matthewmcnew/archtest v0.0.0-20191014222827-a111193b50ad
-	github.com/mattn/go-colorable v0.1.8 // indirect
 	github.com/mgutz/ansi v0.0.0-20170206155736-9520e82c474b
 	github.com/moby/term v0.0.0-20201216013528-df9cb8a40635 // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/sabhiram/go-gitignore v0.0.0-20201211210132-54b8a0bf510f
 	github.com/sclevine/spec v1.4.0
+	github.com/sigstore/cosign v1.0.0
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
 	github.com/theupdateframework/notary v0.6.2-0.20200804143915-84287fd8df4f
 	github.com/vdemeester/k8s-pkg-credentialprovider v1.19.7
 	go.uber.org/zap v1.18.1
-	golang.org/x/crypto v0.0.0-20210513164829-c07d793c2f9a
+	golang.org/x/crypto v0.0.0-20210711020723-a769d52b0f97
 	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
-	k8s.io/api v0.20.7
-	k8s.io/apimachinery v0.20.7
-	k8s.io/client-go v0.20.7
+	k8s.io/api v0.21.3
+	k8s.io/apimachinery v0.21.3
+	k8s.io/client-go v0.21.3
 	k8s.io/code-generator v0.20.7
-	k8s.io/kube-openapi v0.0.0-20210113233702-8566a335510f
+	k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
 	knative.dev/pkg v0.0.0-20210819054404-bda81c029160
 )
+
+replace github.com/prometheus/common => github.com/prometheus/common v0.26.0
```

Comment:
Asking for assistance and insight:

go.mod: with the introduction of cosign, there are different version of dependencies that were indirectly imported. We currently are using a replace to force them to a version. When we run go mod tidy or anything else, it generally updates the versions in the go.mod. For example, k8s.oi/api is v0.20.7, however the previous command updates it to v0.21.3. Is there a way to prevent or improve this?

Reply:
I think we will need the replace directive as you've done here to force:

```
k8s.io/api v0.20.7
k8s.io/apimachinery v0.20.7
k8s.io/client-go v0.20.7
```

but you can group the `replace` like `require`:

```
replace (
  k8s.io/client-go => k8s.io/client-go v0.20.7
  replace k8s.io/api => k8s.io/api v0.20.7
  replace github.com/tj/assert => github.com/tj/assert v0.0.3
)
```

Reply:
created https://github.com/sigstore/cosign/issues/651